File: src-tauri/src/commands/logging.rs
use crate::services::global;
use crate::utils::logger::{LogEntry, GLOBAL_LOG_COLLECTOR};
use tauri::command;

#[command]
pub fn get_logs(limit: Option<usize>) -> Vec<LogEntry> {
    GLOBAL_LOG_COLLECTOR.get_logs(limit)
}

#[command]
pub fn clear_logs() {
    GLOBAL_LOG_COLLECTOR.clear();
}

#[command]
pub fn check_developer_mode() -> bool {
    global::settings_manager().get().developer_mode
}

File: src/api/logging.ts
import { invoke } from "@tauri-apps/api/core";

export interface LogEntry {
  timestamp: string;
  level: string;
  message: string;
}

export async function getLogs(limit?: number): Promise<LogEntry[]> {
  return invoke("get_logs", { limit });
}

export async function clearLogs(): Promise<void> {
  return invoke("clear_logs");
}

export async function checkDeveloperMode(): Promise<boolean> {
  return invoke("check_developer_mode");
}
